server/jetstream.go

```diff
@@ -145,11 +145,17 @@ const (
 	// JetStreamUpdateMsgWithRevision is used to update a message set with a new message iff revision/sequence matches last one.
 	JetStreamUpdateMsgWithRevision = "$JS.%s.UPDATE"
 
-	// JetStreamObservableAckSamplePre is the prefix for sample messages from observables
-	JetStreamObservableAckSamplePre = "$JS.SAMPLE.OBS_ACK"
+	// JetStreamEventPrefix is a prefix for all JetStream events
+	JetStreamEventPrefix = "$JS.EVENT"
 
-	// JetStreamObservableMaxDeliverExceedSamplePre is the prefix for events notifying that a message exceeded max delivery
-	JetStreamObservableMaxDeliverExceedSamplePre = "$JS.EVENT.MAX_DELIVER"
+	// JetStreamSamplePrefix is a prefix for all JetStream samples
+	JetStreamSamplePrefix = "$JS.SAMPLE"
```

Comment:
Not a fan of sample TBH. Its a METRIC or simple EV IMO.

Reply:
So I do like the type separation, there's no good reason for a monitoring subsystem like prom to get metadata about DLQ messages.  So being able to just go $JS.METRIC.> makes sense, in general need to consider separation of concerns as some people might not want to send certain sensitive events over a subject space with other things.

So taking in consideration your other comments, can we say:

`$JS.EV.CONSUMER.MAX_DELIVER`
`$JS.METRIC.CONSUMER.ACK` (or even shorten METRIC)

We can split hairs about these 2 particular events being both events - but I think for the sake of discussion lets assume we will for sure get events of clearer different types, and what I am looking to resolve here is the general design and perhaps not nit pick just on these 2 events.

So lets assume we will have different types of event like messages, some perhaps with sensitive information, bifurcation of subjects will be needed for acls, lets think how that might look using this, perhaps imperfect, example.

Reply:
Sounds good. And looking at it again, lets go with $JS.EVENT.

I have a plan for huffman encoding subjects and accounts, so let's leave them more verbose and fix the bandwidth issues properly like we should.. ;)